### DISCUSSION TITLE
'corosync-cmapctl' enhancement

### DISCUSSION THREAD

**Discussion by angelabriel** [trento-project/wanda#138] [Q&A] [answered]
Today the gatherer 'corosync-cmapctl' accepts as an argument only the whole string representing an entry.
That's absolutely fine for entries like 'totem.join', which have a fixed 'name'. But for entries like 'totem.interface.0.bindnetaddr' it's a bit challenging.
Yes of course we can use 2 facts gathering 'totem.interface.0.bindnetaddr' and 'totem.interface.1.bindnetaddr' and also it will be possible to handle the situation, that on one platform we recommend 2 interfaces (rings) and on another platform 1 interface (ring) will be sufficient.
But it will be much easier to achieve the same goal, if we just can use 'totem.interface' to get the whole matching entries and we can use some Rhai expressions in the 'expectations:' section  to filter and count the needed information (here - number of rings/interfaces)
Will it be possible to enhance the 'corosync-cmapctl' gatherer in this way?

**Comment by nelsonkopliku** [comment, marked as answer]
That's a known behavior indeed. We have an item tracking that need and will be addressed shortly.
Providing `totem.interface` as an argument to the gatherer is going to return a list of entries representing the interfaces detected in the command output.

**Reply by angelabriel** [reply]
many thanks, really appreciated. Will make our 'check live' a little bit easier :smile:
